### ISSUE TITLE
False positive: Add #include <algorithm> for sort

### ISSUE THREAD

**Issue by rakhimov** [cpplint/cpplint#27]
Cpplint seems to assume that all standard algorithms come from the STL,
so, for example, if Boost algorithms are used,
the linter suggests to include `<algorithm>`.
Note that the function call is fully qualified as is required by the Google style:

``` cpp
#include <boost/range/algorithm.hpp>

void foo() {
  boost::sort(std::vector<int>{1, 2, 3, 4});
}
```

The solution strategy could be to take into account `std::` qualification for function calls.

**Comment by tkruse** [comment]
This repo is an "unofficial" fork of cpplint (still the one on pypi though), so you may want to report on https://github.com/google/styleguide.

**Comment by rakhimov** [comment]
@tkruse Good idea. Thanks!
I think I will just reference this pr to get more attention to this "unofficial" but awesome fork.

**Comment by ka-bo** [comment]
I can confirm this behavior with a member function called `transform`.

**Comment by wp-seth** [comment]
the official cpplint is obsolete (see https://github.com/cpplint/cpplint/issues/148#issuecomment-665534852), so it would be great, if that bug could be fixed in this modern repository.

i had the same problem now, for a member function:
```
size_t copy(size_t) {
	return 5;
}
```
where the cpplint message
`Add #include <algorithm> for copy  [build/include_what_you_use] [4]`
just does not make sense.

**Comment by tkruse** [comment]
I would be happy to accept PRs fixing this behavior.
